src/commands/github-bot-comment-on-pr.yml

```diff
@@ -29,5 +36,29 @@ steps:
 
         echo "## << parameters.title >>" > /tmp/preamble.txt
         cat /tmp/preamble.txt /tmp/notification.txt > /tmp/constructed_forslack.txt
-        gh pr comment "$CIRCLE_PULL_REQUEST" -F /tmp/constructed_forslack.txt
+
+        PR_NUMBER=$(echo "$CIRCLE_PULL_REQUEST" | cut -d '/' -f 7)
+        echo $CIRCLE_PULL_REQUEST $PR_NUMBER
+
+        COMMENT_IDS=`gh api repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/issues/$PR_NUMBER/comments \
+          -q '.[] | select(.user.login == "<< parameters.bot-making-comments >>") | select(.body | startswith("## << parameters.title >>")) | .id'`
+        COMMENT_IDS_ARRAY=( $COMMENT_IDS )
+        LAST=""
+        if [ ${#COMMENT_IDS_ARRAY[@]} -ne 0 ] && [ "<< parameters.update-last >>" == "true" ]; then
+            LAST=${COMMENT_IDS_ARRAY[${#COMMENT_IDS_ARRAY[@]}-1]}
+        fi
+
+        if [ "$LAST" == "" ]; then
+            echo No existing comment, create a new one
+            gh pr comment "$CIRCLE_PULL_REQUEST" -F /tmp/constructed_forslack.txt
+        else
+            echo comment exists, update: $LAST
+            body=`cat /tmp/constructed_forslack.txt`
+            gh api --method PATCH \
+                -H "Accept: application/vnd.github+json" \
+                -H "X-GitHub-Api-Version: 2022-11-28" \
+                /repos/$CIRCLE_PROJECT_USERNAME/$CIRCLE_PROJECT_REPONAME/issues/comments/$LAST \
+                -f body="$body"
```

Comment:
_can_ you use the `-F` flag here, instead of reading and dealing with the string in bash?

Reply:
I tried but it didn't work :(
I think `-F` might only work for the things that the github cli supports directly? At least thats my theory.

Reply:
boo. Oh well we tried